halo2_backend/src/plonk/evaluation.rs

```diff
@@ -447,12 +454,13 @@ impl<C: CurveAffine> Evaluator<C> {
                         // - z_i(X) \prod_j (p(X) + \delta^j \beta X + \gamma)
                         // )
                         let mut current_delta = delta_start * beta_term;
-                        for ((set, columns), cosets) in sets
+                        for (((set_idx, _), columns), cosets) in sets
```

Comment:
How about iterating over `permutation_product_cosets` so that we don't need to manually index at each iteration?  Something like this:

```rust
                        let mut current_delta = delta_start * beta_term;
                        for ((permutation_product_coset, columns), cosets) in permutation_product_cosets
                            .iter()
                            .zip(p.columns.chunks(chunk_len))
                            .zip(pk.permutation.cosets.chunks(chunk_len))
                        {
                            let mut left = permutation_product_coset[r_next];
```

I think the code looks slightly more clear this way.